### PR TITLE
Refactor nijel admin backend code

### PR DESCRIPTION
**Pull request by josephbosire**
This PR applies the following changes
 - Move all controllers relating to admin section into controllers/admin dir
 - Extract controllers for different sections from the admin-dashboard.js controller
 - Separate views for different section from the admin-dashboard.pug template
 - Change layout for the projects section to a table view
 - Implements bookmark-able urls for the different admin sections
 - Implement FEATURE A PROJECT functionality
 - implements search for the project section

Tested locally by running the website app using gulp and navigating to admin sections.

### CODE REVIEW

**Comment by josephbosire** [comment]
Still in progress

File: app/js/app.js
Comment (teddyotieno) [inline app/js/app.js:104]: For this particular state, we don't need a templateUrl and its associated view.

File: app/js/controllers/admin/process.js
Comment (teddyotieno) [inline app/js/controllers/admin/process.js]: I like the way you have done the same implementation for the projects part i.e

`getProjects();`

```
   function getProjects(){
        ClientDataService.fetchProjects()
             .then((resp) => {
                 $scope.projects = resp.data.projects;
                 $scope.filteredProjects = resp.data.projects;
             }, (err) => {
                 console.log(err, 'ERROR');
             });
     }
```
its seems more cleaner and easier to understand

File: app/js/controllers/admin/testimonial.js
Comment (teddyotieno) [inline app/js/controllers/admin/testimonial.js]: Not sure where this $scope.variable.selected is being used, I haven't seen it being used anywhere within the controller and view.

File: app/js/controllers/admin/project.js
Comment (teddyotieno) [inline app/js/controllers/admin/project.js]: console statement alert ❗️

File: app/js/controllers/admin/project.js
Comment (teddyotieno) [inline app/js/controllers/admin/project.js]: console statement alert ❗️

File: app/js/controllers/admin/why-nijel.js
Comment (teddyotieno) [inline app/js/controllers/admin/why-nijel.js]: Same again, here $scope.selected isn't being used

File: app/js/controllers/admin/team.js
Comment (teddyotieno) [inline app/js/controllers/admin/team.js]: $scope.selected is here as well

File: app/js/controllers/admin/project.js
Comment (teddyotieno) [inline app/js/controllers/admin/project.js]: Console log statement alert ❗️

File: app/js/controllers/admin/team.js
Comment (teddyotieno) [inline app/js/controllers/admin/team.js]: Again here, I feel the project implementation was better for fetching the projects and adding them to the scope.

File: app/js/app.js
Comment (josephbosire) [reply app/js/app.js:104]: This template forms the base template for the admin section hence the URL

File: app/js/controllers/admin/why-nijel.js
Comment (josephbosire) [reply app/js/controllers/admin/why-nijel.js]: Done. Made the change to remove the unused variable

File: app/js/controllers/admin/testimonial.js
Comment (josephbosire) [reply app/js/controllers/admin/testimonial.js]: Done. Made the change to remove the unused variable

**Comment by josephbosire** [comment]
@judithachieng  and @teddyotieno  Waiting on approvals/feedback so that I can I merge this.

**Review by teddyotieno — APPROVED** [review]

**Review by judithachieng — APPROVED** [review]

**Pull request merged**
